mitodl/mitxpro
# Added user/profile models, serializers, and read-only api

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #45 
Fixes #6 

#### What's this PR do?

This adds a user api and oauth toolkit to enable integration with open edx.

#### How should this be manually tested?

- Follow the setup instructions in `docs/configure_open_edx.md`.
- Login to xPro via /admin
- In OAuth toolkit, create a refresh token and access token for the edx application for your user
- Run this curl request, passing your username, name, email, and access token:
```
curl \
  --data-urlencode "username=$YOUR_USERNAME" \
  --data-urlencode "email=$YOUR_EMAIL" \
  --data-urlencode "name=$YOUR_NAME" \
  --data-urlencode "country=US" \
  --data-urlencode "honor_code=True" \
  --data-urlencode "provider=mitxpro-oauth2" \
  --data-urlencode "access_token=$ACCESS_TOKEN" \
  http://openedx.dev.local:18000/user_api/v1/account/registration/
```
- Go to edx and you should be able to successfully login via 

#### What GIF best describes this PR or how it makes you feel?
![](https://media.giphy.com/media/SiKqz6IPADLF6IvONF/giphy.gif)
